core/src/main/scala/org/apache/spark/scheduler/cluster/CoarseGrainedSchedulerBackend.scala

```diff
@@ -102,6 +104,15 @@ class CoarseGrainedSchedulerBackend(scheduler: TaskSchedulerImpl, val rpcEnv: Rp
   // Executors which are being decommissioned. Maps from executorId to ExecutorDecommissionInfo.
   protected val executorsPendingDecommission = new HashMap[String, ExecutorDecommissionInfo]
 
+  // Unknown Executors which are being decommissioned. This could be caused by unregistered executor
+  // This executor should be decommissioned after registration.
+  // Maps from executorId to (ExecutorDecommissionInfo, adjustTargetNumExecutors,
+  // triggeredByExecutor).
+  protected val unKnownExecutorsPendingDecommission =
+    CacheBuilder.newBuilder()
+      .maximumSize(1000)
```

Comment:
This looks like a random magic number. Do you mean that one `CoarseGrainedSchedulerBackend` can have many un-registered executors like `1000`?

Reply:
In an cluster with spot VM, the number of such executors could be large. What's your suggestion for this? Making this smaller or making it configurable? I plan to keep them at most 5 mins.